submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/service/DoiService.kt

```diff
@@ -0,0 +1,125 @@
+import ac.uk.ebi.biostd.submission.exceptions.InvalidAuthorAffiliationException
+import ac.uk.ebi.biostd.submission.exceptions.InvalidAuthorNameException
+import ac.uk.ebi.biostd.submission.exceptions.InvalidDoiException
+import ac.uk.ebi.biostd.submission.exceptions.InvalidOrgException
+import ac.uk.ebi.biostd.submission.exceptions.InvalidOrgNamesException
+import ac.uk.ebi.biostd.submission.exceptions.MissingAuthorAffiliationException
+import ac.uk.ebi.biostd.submission.exceptions.MissingDoiFieldException
+import ac.uk.ebi.biostd.submission.exceptions.MissingTitleException
+import ac.uk.ebi.biostd.submission.model.Contributor
+import ac.uk.ebi.biostd.submission.model.DoiRequest
+import ac.uk.ebi.biostd.submission.model.SubmitRequest
+import ebi.ac.uk.commons.http.ext.RequestParams
+import ebi.ac.uk.commons.http.ext.post
+import ebi.ac.uk.io.FileUtils
+import ebi.ac.uk.model.Section
+import ebi.ac.uk.model.Submission
+import ebi.ac.uk.model.constants.SubFields.DOI
+import ebi.ac.uk.model.extensions.allSections
+import ebi.ac.uk.model.extensions.title
+import ebi.ac.uk.util.collections.ifNotEmpty
+import mu.KotlinLogging
+import org.springframework.core.io.FileSystemResource
+import org.springframework.http.HttpHeaders
+import org.springframework.http.MediaType.MULTIPART_FORM_DATA
+import org.springframework.util.LinkedMultiValueMap
+import org.springframework.web.reactive.function.client.WebClient
+import java.nio.file.Files
+
+private val logger = KotlinLogging.logger {}
+
+@Suppress("ThrowsCount", "ReturnCount")
+class DoiService(
+    private val webClient: WebClient,
+    private val properties: DoiProperties,
+) {
+    fun calculateDoi(accNo: String, rqt: SubmitRequest): String? {
+        val doi = rqt.submission.attributes.find { it.name == DOI.name } ?: return null
+        val previousDoi = rqt.previousVersion?.doi
+
+        if (previousDoi != null) {
+            require(doi.value == previousDoi) { throw InvalidDoiException() }
+            return previousDoi
+        }
+
+        return registerDoi(accNo, rqt)
+    }
+
+    private fun registerDoi(accNo: String, rqt: SubmitRequest): String {
+        val sub = rqt.submission
+        val title = requireNotNull(sub.title) { throw MissingTitleException() }
+        val request = DoiRequest(accNo, title, properties.uiUrl, getContributors(sub))
+        val requestFile = Files.createTempFile("${TEMP_FILE_NAME}_$accNo", ".xml").toFile()
+        FileUtils.writeContent(requestFile, request.asXmlRequest())
+
+        val headers = HttpHeaders().apply { contentType = MULTIPART_FORM_DATA }
+        val body = LinkedMultiValueMap<String, Any>().apply {
+            add(USER_PARAM, properties.user)
+            add(PASSWORD_PARAM, properties.password)
+            add(OPERATION_PARAM, OPERATION_PARAM_VALUE)
+            add(FILE_PARAM, FileSystemResource(requestFile))
+        }
+
+        webClient.post(properties.endpoint, RequestParams(headers, body))
```

Comment:
I would recomend to use functions so it becomes 
```
        val headers = httpHeadersOf(contentType to MULTIPART_FORM_DATA)
        val body = linkedMultiValueMapOf(
            USER_PARAM to properties.user,
            PASSWORD_PARAM to properties.password,
            OPERATION_PARAM to OPERATION_PARAM_VALUE,
            FILE_PARAM to FileSystemResource(requestFile)
        )  
```

Reply:
This is a great functionality but it's part of the `bio-webclient` module which is not used in the `submitter` module and I think it shouldn't be. Instead, I moved this method to `commons-http` which seems like a best place for this general functionality to be in so it can be used in every module